src/common/AdaptiveStream.cpp

```diff
@@ -1055,22 +1051,24 @@ bool AdaptiveStream::seek_time(double seek_seconds, bool preceeding, bool& needR
   if (newSeg)
   {
     needReset = true;
-    if (newSeg != old_seg)
+    if ((newSeg != old_seg) || (!preceeding && state_ == STOPPED))
     {
       StopWorker(STOPPED);
       // EnsureSegment loads always the next segment, so go back 1
       current_rep_->current_segment_ =
           current_rep_->get_segment(current_rep_->get_segment_pos(newSeg) - 1);
       // TODO: if new segment is already prefetched, don't ResetActiveBuffer;
       ResetActiveBuffer(false);
+      if (newSeg == old_seg && !preceeding)
+      {
+        absolute_position_ -= segment_read_pos_;
+        segment_read_pos_ = 0;
+      }
     }
-    else if (!preceeding)
+    else if (preceeding)
     {
-      absolute_position_ -= segment_read_pos_;
-      segment_read_pos_ = 0;
-    }
-    else
       needReset = false;
+    }
```

Comment:
Here the goal is to make sure that if the state is stopped we start it again. Not needed to check state when preceeding is only true when seeking across periods (not sure why this is btw)

On another note, `StopWorker` is horribly named

Reply:
yeah the name is inappropriate